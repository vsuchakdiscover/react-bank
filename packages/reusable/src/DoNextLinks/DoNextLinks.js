import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import Button from "../Button";
import PrintButton from "../Button/PrintButton";
import SmartLink from "../SmartLink";
import cx from "classnames";
import styles from "./DoNextLinks.module.scss";

const DoNextLinks = ({ header, showHeader, className, children, ...props }) => (
  <div className={cx(styles.root, className)} {...props}>
    {showHeader && <p>{header}</p>}
    <div className={styles.links}>{children}</div>
  </div>
);

DoNextLinks.propTypes = {
  /** Assure children are of valid type */
  checkChildren: function (props, propName, componentName) {
    let error = null;
    props.children.forEach(function (child) {
      if (
        child.type !== "a" &&
        child.type !== "button" &&
        child.type !== Button &&
        child.type !== Link &&
        child.type !== PrintButton &&
        child.type !== SmartLink &&
        child.type !== undefined
      ) {
        error = true;
      }
    });
    if (error) {
      return new Error(
        `Invalid prop ${propName} supplied to '${componentName}'. Validation failed.`
      );
    }
  },

  /** Child elements -only anchors or buttons- to render inside button */
  children: PropTypes.any.isRequired,

  /** Class to apply */
  className: PropTypes.string,

  /** Header Text */
  header: PropTypes.string.isRequired,

  /** Set to false if header should be hidden */
  showHeader: PropTypes.bool,
};

DoNextLinks.defaultProps = {
  header: "What would you like to do next?",
  showHeader: true,
};

export default DoNextLinks;
