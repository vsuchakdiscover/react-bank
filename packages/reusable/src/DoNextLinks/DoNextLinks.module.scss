@import "src/scss/variables";

.root {
  font-family: "Meta Web Normal";
  font-weight: normal;
  p {
    margin-bottom: 5px;
    font-family: "Meta Web Bold";
  }
  a,
  button {
    color: $links;
    position: relative;
    display: block;
    padding: 10px 0;
    margin-bottom: 10px;
    line-height: 15px;
    border-radius: 0;
    @media #{$tablet} {
      display: inline-block;
      padding: 10px 19px;
      margin-bottom: 0;
      &:after {
        display: block;
        content: "";
        position: absolute;
        height: 20px;
        width: 2px;
        background: $light-gray;
        top: 6px;
        right: 0px;
      }
      &:last-child {
        &:after {
          display: none;
        }
      }
    }
  }
}

.links {
  display: inline-block;
}

.links a:first-child {
  @media #{$tablet} {
    padding-left: 0;
  }
}
