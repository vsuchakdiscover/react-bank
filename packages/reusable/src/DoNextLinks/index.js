// Barrel for shortening imports
export { default } from "./DoNextLinks";
