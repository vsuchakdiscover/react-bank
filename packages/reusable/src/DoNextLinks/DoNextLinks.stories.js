import React from "react";
import DoNextLinks from "./DoNextLinks";
import { BrowserRouter as Router } from "react-router-dom";
import Button, { BUTTON_TYPES } from "../Button";
import { text } from "@storybook/addon-knobs";

export default {
  title: "Links/Do Next Links",
  parameters: {
    info: {
      propTables: [DoNextLinks],
    },
  },
};

export const defaultExample = () => (
  <Router>
    <DoNextLinks>
      <a href="#makeABillPayment">Make a Bill Payment</a>
      <a href="#managePayees">Regular Link</a>
    </DoNextLinks>
  </Router>
);

export const custom = () => (
  <DoNextLinks header={text("Header text", "What would you like for dinner?")}>
    <a href="#Spaghett">Spaghett</a>
    <a href="#Tacos">Tacos</a>
    <a href="#Boiled Ham">Boiled Ham</a>
    <Button buttonStyle={BUTTON_TYPES.LINK}>A button could be here too!</Button>
  </DoNextLinks>
);
