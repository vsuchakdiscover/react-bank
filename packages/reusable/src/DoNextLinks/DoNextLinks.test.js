import React from "react";
import DoNextLinks from "./DoNextLinks";
import renderer from "react-test-renderer";

it("renders with the default header", () => {
  const tree = renderer
    .create(
      <DoNextLinks>
        <a href="#makeABillPayment">Make a Bill Payment</a>
        <a href="#managePayees">Manage Payees</a>
      </DoNextLinks>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders with a custom header", () => {
  const tree = renderer
    .create(
      <DoNextLinks header="Custom Header">
        <a href="#makeABillPayment">Make a Bill Payment</a>
        <a href="#managePayees">Manage Payees</a>
      </DoNextLinks>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
