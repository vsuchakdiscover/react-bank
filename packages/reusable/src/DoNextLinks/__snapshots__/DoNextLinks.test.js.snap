// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with a custom header 1`] = `
<div
  className="root"
>
  <p>
    Custom Header
  </p>
  <div
    className="links"
  >
    <a
      href="#makeABillPayment"
    >
      Make a Bill Payment
    </a>
    <a
      href="#managePayees"
    >
      Manage Payees
    </a>
  </div>
</div>
`;

exports[`renders with the default header 1`] = `
<div
  className="root"
>
  <p>
    What would you like to do next?
  </p>
  <div
    className="links"
  >
    <a
      href="#makeABillPayment"
    >
      Make a Bill Payment
    </a>
    <a
      href="#managePayees"
    >
      Manage Payees
    </a>
  </div>
</div>
`;
